brave/src/main/java/brave/propagation/ExtraField.java

```diff
@@ -0,0 +1,330 @@
+ * countryCode = ExtraField.create("country-code");
+ *
+ * // If configured and set, you can retrieve it later. All of the below result in the same tag:
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue());
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue(context));
+ * spanCustomizer.tag("country-code", ExtraField.getValue("country-code"));
+ *
+ * // You can also set or override the value similarly, which might be needed if a new request
+ * countryCode.setValue("FO");
+ * countryCode.setValue(context, "FO");
+ * ExtraField.setValue("country-code", "FO");
+ * ExtraField.setValue(context, "country-code", "FO");
+ * }</pre>
+ *
+ * <h3>Correlation</h3>
+ * If you want an extra field to also be available in correlation such as logging contexts, use
+ * {@link Builder#withCorrelation()}.
+ *
+ * <pre>{@code
+ * // configure the field, permitting it to be used in correlation contexts
+ * amznTraceId = ExtraField.newBuilder("x-amzn-trace-id").withCorrelation().build();
+ *
+ * // Allow logging patterns like %X{traceId} %X{x-amzn-trace-id}
+ * loggingContext = new Log4J2Context();
+ *
+ * decorator = CorrelationFieldScopeDecorator.newBuilder(loggingContext)
+ *                                           .addField(amznTraceId).build();
+ *
+ * tracingBuilder.propagationFactory(ExtraFieldPropagation.newFactoryBuilder(B3Propagation.FACTORY)
+ *                                                        .addField(amznTraceId)
+ *                                                        .build())
+ *               .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+ *                                                                  .addScopeDecorator(decorator)
+ *                                                                  .build())
+ * }</pre>
+ *
+ * <h3>Prefixed fields</h3>
+ * <p>You can also prefix fields, if they follow a common pattern. For example, the following will
+ * propagate the field "x-vcap-request-id" as-is, but send the fields "country-code" and "user-id"
+ * on the wire as "baggage-country-code" and "baggage-user-id" respectively.
+ *
+ * <pre>{@code
+ * requestId = ExtraField.create("x-vcap-request-id");
+ * userId = ExtraField.newBuilder("user-id").prefix("baggage-").build();
+ * countryCode = ExtraField.newBuilder("country-code").prefix("baggage-").build();
+ *
+ * // Later, you can call below to affect the country code of the current trace context
+ * ExtraField.setValue("country-code", "FO");
+ * String countryCode = ExtraField.getValue("country-code");
+ * }</pre>
+ *
+ * <h3>Appropriate usage</h3>
+ * It is generally not a good idea to use the tracing system for application logic or critical code
+ * such as security context propagation.
+ *
+ * <p>Brave is an infrastructure library: you will create lock-in if you expose its apis into
+ * business code. Prefer exposing your own types for utility functions that use this class as this
+ * will insulate you from lock-in.
```

Comment:
this is already mentioned in brave's main README.md , any merit repeating the obvious here ? The security bits are very relevant but this applies to all public types in brave no ?

Reply:
The reason this is in the type doc is that this type specifically adds temptation to embed in your code, and readme doesn't inherit docs and visa versa (maybe asciidoc can help if someone can help with it)

That said, this doc was ignored on the type it was on last time, leading to this very change, heh!

Reply:
gotcha.